packages/stack/blockchain/src/index.js

```diff
@@ -22,57 +21,50 @@ class Blockchain {
     embark.registerActionForEvent("pipeline:generateAll:before", this.addArtifactFile.bind(this));
 
     this.blockchainNodes = {};
-    this.events.setCommandHandler("blockchain:node:register", (clientName, startCb) => {
-      this.blockchainNodes[clientName] = startCb;
+    this.events.setCommandHandler("blockchain:node:register", (clientName, { isStartedFn, launchFn, stopFn }) => {
+
+      if (!isStartedFn) {
+        throw new Error(`Blockchain client '${clientName}' must be registered with an 'isStarted' function, client not registered.`);
+      }
+      if (!launchFn) {
+        throw new Error(`Blockchain client '${clientName}' must be registered with a 'launchFn' function, client not registered.`);
+      }
+      if (!stopFn) {
+        throw new Error(`Blockchain client '${clientName}' must be registered with a 'stopFn' function, client not registered.`);
+      }
```

Comment:
@emizzle I've updated this PR here. `setCommandHandler()` doesn't have a notion of error callback so we have to throw here. In the future, this ideally gets caught by `tsc` at compile time.